### PR TITLE
[JENKINS-42505] Improve test for config update/removal

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-42505](https://issues.jenkins-ci.org/browse/JENKINS-42505)

- Include a new config to verify the proper one is updated.
- Assert the removed config is the expected one.

@reviewbybees @imod

### CODE REVIEW

**Comment by imod** [comment]
any one from cloudbees?

File: src/test/java/org/jenkinsci/plugins/configfiles/ConfigFileStoreTest.java [hunk 1]
```diff
@@ -34,14 +32,29 @@ public void testSaveGlobalConfigFiles() {
...
+        Assert.assertEquals(2, store.getConfigs().size());
+        savedConfig = store.getById(CONFIG_ID);
         Assert.assertEquals("new name", savedConfig.name);
         Assert.assertEquals("new comment", savedConfig.comment);
         Assert.assertEquals("new content", savedConfig.content);
+
+        // Remove config. Check the correct one is removed
+        store.remove(savedConfig.id);
+        Assert.assertEquals(1, store.getConfigs().size());
+        Assert.assertThat(store.getById(anotherConfig.id), Matchers.is((Config) anotherConfig));
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/configfiles/ConfigFileStoreTest.java]: or

```java
Matchers.<Config>is(anotherConfig))
```

**Comment by varyvol** [comment]
@jglick thanks.

**Comment by mafraba** [comment]
🐝

**Pull request merged**